### PR TITLE
Remove WellKnownType.Last

### PR DESCRIPTION
**Pull request by cston**
Remove `WellKnownType.Last` and rename `WellKnownType.Available` to `NextAvailable`.

### CODE REVIEW

**Comment by cston** [comment]
@dotnet/roslyn-compiler please review.

**Comment by jcouv** [comment]
LGTM

File: src/Compilers/Core/Portable/WellKnownTypes.cs [hunk 1]
```diff
@@ -541,10 +540,6 @@ private static void AssertEnumAndTableInSync()
                 {
                     typeIdName = "System.Math";
                 }
-                else if (typeId == WellKnownType.Last)
```
Comment (agocke) [inline src/Compilers/Core/Portable/WellKnownTypes.cs]: Do we still need this? Is it used as an index into the names?

**Comment by gafter** [comment]
LGTM

File: src/Compilers/Core/Portable/WellKnownTypes.cs (hunk 1 above)
Comment (cston) [inline src/Compilers/Core/Portable/WellKnownTypes.cs]: @agocke `WellKnownType.Last` was only used for range checking, not as an index.

**Pull request merged**